internal/service/advancedclustertpf/resource.go

```diff
@@ -348,7 +384,6 @@ func (r *rs) updateLegacyReplicationSpecs(ctx context.Context, state, plan *TFMo
 		return false
 	}
 	api20240530 := r.Client.AtlasV220240530.ClustersApi
-	api20240530.UpdateCluster(ctx, plan.ProjectID.ValueString(), plan.Name.ValueString(), legacyPatch)
```

Comment:
@EspenAlbert this seems a leftover, it's just the API request without execute, next line it's the same with execute